solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -89,14 +93,61 @@ public static Health combine(Collection<Health> states) {
     }
   }
 
-  public ClusterStatus(ZkStateReader zkStateReader, ZkNodeProps props) {
+  public ClusterStatus(ZkStateReader zkStateReader, SolrParams params) {
     this.zkStateReader = zkStateReader;
-    this.message = props;
-    collection = props.getStr(ZkStateReader.COLLECTION_PROP);
+    this.solrParams = params;
+    collection = params.get(ZkStateReader.COLLECTION_PROP);
   }
 
   public void getClusterStatus(NamedList<Object> results)
       throws KeeperException, InterruptedException {
+    NamedList<Object> clusterStatus = new SimpleOrderedMap<>();
+
+    boolean includeAll = solrParams.getBool(INCLUDE_ALL, true);
+    boolean withLiveNodes = solrParams.getBool(LIVENODES_PROP, includeAll);
+    boolean withClusterProperties = solrParams.getBool(CLUSTER_PROP, includeAll);
+    boolean withRoles = solrParams.getBool(ZkStateReader.ROLES_PROP, includeAll);
+    boolean withCollection = includeAll || (collection != null);
+
+    List<String> liveNodes = Collections.emptyList();
```

Comment:
shouldn't the default be null since it won't necessarily be fetched?

Reply:
Updated. I had it as an empty list earlier to avoid a NPE further on, but the assert achieves the same purpose